livedata/src/main/java/io/getstream/chat/android/livedata/entity/MemberEntity.kt

```diff
@@ -36,15 +37,8 @@ data class MemberEntity(var userId: String) {
     }
 
     /** converts a member entity into a member */
-    fun toMember(userMap: Map<String, User>): Member {
-        val user = userMap[userId] ?: error("userMap is missing the user for this member")
-        val r = Member(user, role)
-        r.createdAt = createdAt
-        r.updatedAt = updatedAt
-        r.isInvited = isInvited
-        r.inviteAcceptedAt = inviteAcceptedAt
-        r.inviteRejectedAt = inviteRejectedAt
-
-        return r
-    }
+    fun toMember(userMap: Map<String, User>): Member? =
+        userMap[userId]?.let {
+            Member(it, role, createdAt, updatedAt, isInvited, inviteAcceptedAt, inviteRejectedAt)
+        } ?: null.also { ChatLogger.get("MemberEntity").logE("userMap is missing the user with id='$userId` needed to create this member") }
```

Comment:
I know it's boring code, but wouldn't this be more readable than `x?.let { } ?: null.also { }`? The `null.also` part is especially making me uncomfortable 😅 

```suggestion
    fun toMember(userMap: Map<String, User>): Member? {
        val user = userMap[userId]
        if (user == null) {
            ChatLogger.get("MemberEntity")
                .logE("userMap is missing the user with id='$userId` needed to create this member")
            return null
        }
        return Member(user, role, createdAt, updatedAt, isInvited, inviteAcceptedAt, inviteRejectedAt)
    }
```

Reply:
I would prefer that Kotlin provides some kind of sideefect extension for Elvis operator, but it is only possible for "sucess path", and the propose code seems "so old java style"..., but well, we can change it

Reply:
It _is_ "Java style", but it's also very easily parseable, anyone looking at it will know what it does immediately, and not have to wonder about what `also` is called on (just `null`, or the entire Elvis expression), or have to think through what happens exactly when something along the way in the chain is `null`.

You're basically unrolling the Kotlin style code into this in your head anyway, which is extra processing 😄

Reply:
👏 simple is better. It's ok to spend a few extra lines for readability.

(coming from someone who used to love abusing python metaclasses for side effects. Nowadays I really like languages like Go which basically don't allow you to get creative. )

That being said, Kotlin can have some issues with nullability and if statements. It doesn't always understand that it's in a clause where the value can't be null.

Reply:
In the cases you're thinking of, the compiler is actually (at least technically) correct. It doesn't give you smart casts in `if` checks if you have a mutable, non-local variable, because it would be unsafe in a multithreaded environment. I've written this down in detail [here](https://stackoverflow.com/a/51094692/4465208).